models/surveys.go

```diff
@@ -486,13 +493,42 @@ func (api *API) Info(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-// AllSurveys returns summaries of all known surveys. The surveys are returned in ascending short name order.
+//AllSurveys handler returns a list of all surveys
 func (api *API) AllSurveys(w http.ResponseWriter, r *http.Request) {
-	rows, err := api.AllSurveysStmt.Query()
+	api.GetSurveys(w, r, "")
+}
 
-	if err != nil {
-		http.Error(w, "AllSurveys query failed", http.StatusInternalServerError)
-		return
+//SurveysByType handler returns a list of all surveys of a particular type e.g Business or Social
+func (api *API) SurveysByType(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	surveyType := vars["surveyType"]
+
+	// Mux package does not pass survey type during tests
+	if surveyType == "" {
+		surveyType = "Business"
+	}
+	api.GetSurveys(w, r, surveyType )
+}
+
+// GetSurveys returns summaries of all known surveys. The surveys are returned in ascending short name order.
+// optional: surveyType if supplied then the list only returns surveys which match that surveyType
+// valid surveyTypes are 'Business', 'Social' or 'Census'
+func (api *API) GetSurveys(w http.ResponseWriter, r *http.Request, surveyType string) {
+	var rows *sql.Rows
+	var err error
+
+	if len(surveyType) == 0 {
```

Comment:
More idiomatic to avoid `else` wherever possible

Could do something more like: (depending on whether `Query()` complains about empty strings in bindings)
```
stmt := api.AllSurveys
if len(surveyType) > 0 {
	stmt = api.GetSurveysBySurveyTypeStmt
}

rows, err = stmt.Query(surveyType)
if err != nil {
	log.Printf("failure retrieving surveys for type '%s'",surveyType) 
	http.Error(w, "failed to retrieve surveys", https.StatusInternalServerError)
	return
}
```

Reply:
👍 for avoiding else in general - not only an idiom, but applied, it makes a functional approach (and so pure functions and thus unit testing) easier in virtually any language.